src/main/java/me/ryanhamshire/GriefPrevention/Claim.java

```diff
@@ -307,51 +311,89 @@ public boolean isNear(Location location, int howNear)
 
         return claim.contains(location, false, true);
     }
-
+    
     //permissions.  note administrative "public" claims have different rules than other claims
     //all of these return NULL when a player has permission, or a String error message when the player doesn't have permission
     public String allowEdit(Player player)
+    {
+        return this.allowEdit(player, true);
+    }
+
+    //permissions.  note administrative "public" claims have different rules than other claims
+    //all of these return NULL when a player has permission, or a String error message when the player doesn't have permission
+    public String allowEdit(Player player, boolean fireEvent)
     {
         //if we don't know who's asking, always say no (i've been told some mods can make this happen somehow)
         if (player == null) return "";
 
-        //special cases...
-
-        //admin claims need adminclaims permission only.
-        if (this.isAdminClaim())
-        {
-            if (player.hasPermission("griefprevention.adminclaims")) return null;
-        }
+        String message;
+        // default system before event firing
+        test: {
```

Comment:
I didn't think at adding "else if" blocks because I wanted to have something as near of the original code as possible but I'm going to fix this (if it's worth it).

Reply:
I've some problems yet : when there is 
```
if(...[1]) {
    if(...[2]) {
        return ... ;
    }
} else { ... }
```
, it is really not the same as 
```
if(...[1] && ...[2]) {
     return ... ;
} else { ... }
```
so I don't know how to do with only else blocks (because there are situations like that, e.g. with the test for admin claims) and it's maybe because I'm a bad developer but I'm still stuck...

Reply:
I think you're probably just getting too hung up on maintaining a minimal line content diff. When you look at the actual logic, there are few else blocks used, which means that all failed conditions fall through to the next if. That means that if you stack all of the conditions in a single block, you can actually achieve the required behavior with an else.

Examine the following:
```Java
if (condition1) {
  if (condition2) {
    return x;
  }
}
if (condition3) {
  return y;
}
return z;
```
Vs.
```Java
if (condition1 && condition2) {
  return x;
} else if (condition3) {
  return y;
} else {
  return z;
}
```

If it's truly not possible to do in an efficient manner without quick returns, then a helper method should be created instead.

Reply:
I think I'll use a helper method because there's an `elseif` with the `condition3` so it is not equivalent (maybe in practice well but I don't want to take the risk).